python/cuml/datasets/blobs.pyx

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import cuml
+import numpy as np
+
+from collections.abc import Sequence
+
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import get_dev_array_ptr, zeros, input_to_dev_array
+
+from libcpp cimport bool
+from libc.stdint cimport uint64_t, uintptr_t
+
+from numba import cuda
+
+from random import randint
+
+cdef extern from "datasets/make_blobs.hpp" namespace "ML::Datasets":
+    cdef void make_blobs(const cumlHandle& handle,
+                         float* out,
+                         int* labels,
+                         int n_rows,
+                         int n_cols,
+                         int n_clusters,
+                         const float* centers,
+                         const float* cluster_std,
+                         const float cluster_std_scalar,
+                         bool shuffle,
+                         float center_box_min,
+                         float center_box_max,
+                         uint64_t seed) except +
+
+    cdef void make_blobs(cumlHandle& handle,
+                         double* out,
+                         int* labels,
+                         int n_rows,
+                         int n_cols,
+                         int n_clusters,
+                         double* centers,
+                         double* cluster_std,
+                         double cluster_std_scalar,
+                         bool shuffle,
+                         double center_box_min,
+                         double center_box_max,
+                         uint64_t seed) except +
+
+inp_to_dtype = {
+    'single': np.float32,
+    'float': np.float32,
+    'double': np.float64,
+    np.float32: np.float32,
+    np.float64: np.float64
+}
+
+
+# Note: named blobs to avoid cython naming conflict issues, renaming in
```

Comment:
The only reason I'm not in love with this solution is because I no longer have the option of using `cuml.datasets.make_blobs`, which corresponds directly to sklearn. I am not completely in love with removing the packaged namespace and providing that as the only option, mostly because I'm used to looking in "cuml.datasets.make_blobs" and it makes an extra assumption about my knowledge of cuml, which as a user just giving it a try, I might not yet have. 

I'm more in favor of changing the name of the prim to be something like `make_blobs_dataset` or `make_blobs_data`, that way we don't have to remove our packaged namespace.

Reply:
Actually you can call it with `cuml.datasets.make_blobs` as well, I thought of that ;)